src/utils.js

```diff
@@ -669,6 +670,60 @@ export const parseContentTypeFromResponse = (response) => {
     };
 };
 
+/**
+ *
+ * @param response
+ * @return {undefined|Array}
+ */
+export const getCookiesFromResponse = (response) => {
+    const { headers } = response;
+    let cookies;
+
+    if (Array.isArray(headers['set-cookie'])) {
+        cookies = headers['set-cookie'].map(Cookie.parse);
+    } else {
+        cookies = [Cookie.parse(headers['set-cookie'])];
+    }
+
+    return cookies;
+};
+
+/**
+ * Gets cookie string for the "Cookie" header
+ * @param cookies {Array}
+ * @return {String|undefined}
+ */
+export const getCookieHeader = (cookies) => {
+    if (cookies.length < 1) {
+        return;
+    }
+    return cookies.map(cookie => cookie && cookie.cookieString()).join(';');
+};
+
+/**
+ * Updates cookies from response to the `Session` instance.
+ * @param session {Session} - `Session` instance to be updated.
+ * @param response
+ * @return {Session}
+ *
+ */
+export const updateSessionCookies = (session, response) => {
+    const newCookies = getCookiesFromResponse(response);
+    const { cookies: oldCookies } = session;
+
+    for (const cookie of newCookies) {
+        const cookieIndex = oldCookies.findIndex(c => c.key === cookie.key);
```

Comment:
I think that we should use the `CookieJar` for this. The cookies are `key` `value` pairs, but there are other variables such as domains, paths, expiry dates etc. We should not be sending cookies to other domains and we should not send old cookies to same domain, it can introduce problems.